### ISSUE TITLE
RFE: GitHub: Refuse merge on "failed" build status

### ISSUE THREAD

**Issue by sils** [yegor256/rultor#869]
At least your docs don't state this feature, I didn't test it yet.

It would be quite cool if rultor would simply check all GitHub build statuses before merging. It is often not possible to run the whole CI stuff on rultor and usually external CI services do the merge before testing. I'd love it if rultor was able to refuse the merge if any build status is unfinished or failed.

**Comment by yegor256** [comment]
@sils1297 that's a very good idea, thanks!

**Comment by alex-palevsky** [comment]
@sils1297 milestone set to `2.0` (correct me if I am wrong)

**Comment by alex-palevsky** [comment]
@yegor256 it is yours,help us

**Comment by sils1297** [comment]
Bonus points: a parameter in the yml to at least have the given number of positive statuses before merging - some take some time to appear. Also all statuses should be succeeded, none must be pending.

**Comment by sils1297** [comment]
I just got an idea. So what would be really really awesome would be if we could just send off `@rultor merge` command and rultor waits on its own until the given number of GitHub statusses is succeeded (and none are pending or failed) and *then* it does the merge or fail.

So after review you can just dispatch the merge and let the CI figure out the rest, not needing to manually check every minute if that damn windows build still is unfinished.

**Comment by yegor256** [comment]
@sils1297 it seems that Github doesn't expose this information through [their API](https://developer.github.com/v3/pulls/)... or I miss something?

**Comment by sils1297** [comment]
How about https://developer.github.com/v3/repos/statuses/ ?

2015-07-15 1:29 GMT+02:00 Yegor Bugayenko <notifications@github.com>:

**Comment by sils1297** [comment]
@yegor256 any progress? This would be a *main* feature of rultor for me.

**Comment by yegor256** [comment]
@sils1297 I missed your previous message. Looks like it all doable, I'll implement it soon

**Comment by sils1297** [comment]
Oh and actually there's one relevant issue that may be an own bug: one should be able to temporary override that kind of lock. Sometimes CI providers have issues on their own (e.g. travis has have problems with their infrastructure recently, scrutinizer has some right now which is why I'm fling that bug and so on.) that mark builds failed or they'll never finish. It may be not acceptable for a development team that relies e.g. on refactorings to continue work to wait for them to resolve this which might take a day or even more. So this override would probably be an optional parameter of the merge command I guess and rultor should shout awful things on usage of this so the user at least feels bad about it unless he can shift the blame to travis and co.

**Comment by yegor256** [comment]
@rultor assign someone else pls

**Comment by yegor256** [comment]
@alex-palevsky assign someone else pls

**Comment by alex-palevsky** [comment]
> @alex-palevsky assign someone else pls

@yegor256 I deducted 30 points from your rating

**Comment by alex-palevsky** [comment]
> @alex-palevsky assign someone else pls

@yegor256 I will try to find someone else

**Comment by original-brownbear** [comment]
@alex-palevsky this is postponed.

**Comment by alex-palevsky** [comment]
> @alex-palevsky this is postponed.

@original-brownbear OK, I put "postponed" label here

**Comment by alex-palevsky** [comment]
> @alex-palevsky this is postponed.

@original-brownbear got it, someone else will be assigned soon

**Comment by pnatashap** [comment]
@yegor256 looks like this issue can be closed: before merging rultor checks all existing autochecks and refuse merging in case any of them are failed

**Comment by yegor256** [comment]
@sils indeed, it works now
